library/proc_macro/src/bridge/client.rs

```diff
@@ -11,9 +11,9 @@ macro_rules! define_handles {
     ) => {
         #[repr(C)]
         #[allow(non_snake_case)]
-        pub struct HandleCounters {
-            $($oty: AtomicUsize,)*
-            $($ity: AtomicUsize,)*
+        pub(super) struct HandleCounters {
+            $(pub(super) $oty: AtomicUsize,)*
+            $(pub(super) $ity: AtomicUsize,)*
```

Comment:
I don't think the fields being public is acceptable. The `FIXME` you went for is likely tied to the fact that it's impossible to have `HandleStore::new` defined anywhere other than in this module.

You could maybe move everything except `HandleStore::new`?

Reply:
Why is it not acceptable? The new division seems better: all the `HandleStore` stuff is in `server.rs`, where it belongs. `HandleStore` does need access to the fields of `HandleCounters`, but that makes sense given that `HandleStore` is layered on top of `HandleCounters`.

(Why is `HandleCounters` on the client side, BTW? Is it so that each proc macro gets its own separate set of counters? Would there be a problem if it was on the server side and all proc macros shared one set of counters?)

Some good news: your comment also made me realize that there was room for improvement in that last commit. I have removed the `pub(super)` markers on `HandleStore`, `HandleStore`'s fields, `HandleStore::new`, `MarkedTypes`. I have also removed all the `server::` qualifiers from the code moved into `server.rs`. Thanks for that!

Reply:
> Why is `HandleCounters` on the client side, BTW?

I just tried moving it into `server.rs` and everything worked fine. All tests pass, and it removed the need for the `pub(super)` markings.

Maybe there's a reason for having it in `client.rs`, but there's no apparent explanation in either the code or the test suite. So I've pushed the commit to this PR in case it's usable.